### DISCUSSION TITLE
Feature Request: Adjustable Volume for "Play an Audio on Copy" in Pano Clipboard Manager

### DISCUSSION THREAD

**Discussion by NoobGajen** [oae/gnome-shell-pano#274] [Ideas]
Is there a way to adjust the volume for the audio played when I copy something using Pano clipboard manager in GNOME? Sometimes, it's a bit too loud for me. Could we add a feature to control the volume level?

**Comment by Totto16** [comment]
At the moment there is no way, but it's an easy to implement feature, that is generally a good idea 👍🏼 
If you want, feel free to add a PR, since this doesn't require that much of a deep knowledge or dive into the existing code. It can be done pretty easily. Otherwise, we might add it ourselves, but atm. we have higher priorities like bugs or GNOME 46 compatibility, since this is a feature, it might take some time

**Comment by NoobGajen** [comment]
I tried, but I couldn't find a way to do this. I was planning to find and replace audio files by reducing their volume, but there are no such audio files. Can you help me tweak the sound level from the code?

**Comment by Totto16** [comment]
We just play the gnome-shell sound for `message`: see  [here](https://github.com/oae/gnome-shell-pano/blob/master/src/utils/shell.ts#L187-L192)

There you can also see, that we use `soundContext.play_simple` [see here](https://gjs-docs.gnome.org/gsound10~1.0/gsound.context#method-play_simplev)

GSound uses attributes to set some properties, see   [here](https://gjs-docs.gnome.org/gsound10~1.0-constants/)

You can use [GSound.ATTR_CANBERRA_VOLUME](https://gjs-docs.gnome.org/gsound10~1.0/gsound.attr_canberra_volume)

To set the volume, the documentation describes the range of it, so the things you would have to do, would be:
- add a setting for the volume
- read the setting there and set the volume attribute


If you need any further information or help, feel free to ask again and open eventually a PR ❤️
